### ISSUE TITLE
.NET Core Identity doesn't refresh auth cookie for static files and throws 500 error

### ISSUE THREAD

**Issue by gladkiy** [aspnet/Identity#1271]
I have .NET Core Web API app with Angular 2 on client-side (also use lazy loading).
I setup .NET Core Identity with the following options in **Startup.cs**:

```
options.Password.RequiredLength = 5;                                              
options.Password.RequireDigit = false;
options.Password.RequireLowercase = false;
options.Password.RequireUppercase = false;
options.Password.RequireNonAlphanumeric = false;
options.SecurityStampValidationInterval = TimeSpan.FromMinutes(1);
options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(5);
options.Lockout.MaxFailedAccessAttempts = 100;
options.Cookies.ApplicationCookie.ExpireTimeSpan = TimeSpan.FromDays(14);
options.Cookies.ApplicationCookie.LoginPath = "/api/signin/signin";
options.Cookies.ApplicationCookie.LogoutPath = "/api/signin/signout";
options.User.RequireUniqueEmail = false;
```

So,
User is logged in - auth cookie is set.
60 seconds later - calls to **API/*** end-points update this cookie to a new value .. so calls work
However calls to **HTML/JS** files do not update the cookie - they continue to use the previous cookie which now is presumably not tied to the session which then means we get the 500 error as it can't do something.
Loading any **API/*** end-point updates the cookie and **HTML/JS** files work again.
So in short - the issue seems to be related to auth cookie not being refreshed for all the static content. And it just works fine without any cookie, when user is logged out.
Also I know that _the static file module provides no authorization checks_. I can't understand why it fails.
When I disable Angular lazy loading it seems issue no longer exists because all **HTML/JS** files is loaded straight away. But I need lazy loading because it speeds up my app.

My Configure method in `Startup.cs`:
```
public void Configure(IApplicationBuilder app, IHostingEnvironment env)
{
    app.UseIdentity();
    app.UseSession();
    app.UseMiddleware<PathLocationStrategyMiddleware>();
    app.UseMiddleware<ErrorMiddleware>();
    app.UseDefaultFiles();
    app.UseStaticFiles();
    app.UseMvc();
}
```

**Comment by blowdart** [comment]
What order is your middleware in? i.e. is static before or after auth?

**Comment by gladkiy** [comment]
@blowdart I've updated my question. Do you mean `app.UseDefaultFiles();` and `app.UseStaticFiles();` need to be over `app.UseIdentity();`?

**Comment by blowdart** [comment]
No, .UseIdentity should be first. Admittedly static files doesn't support auth in any way, but the cookie should still refresh imo. @HaoK any ideas?

**Comment by HaoK** [comment]
The cookie is being refreshed by the security stamp validator, I don't think static files would interfere with that, @Tratcher does this issue sound familiar to you at all?

**Comment by HaoK** [comment]
Is your static content cached on the client maybe?

**Comment by Tratcher** [comment]
The 500 is caused by an exception on the server. Check the logs to see what the exception was.

**Comment by gladkiy** [comment]
May be it's cache, I don't know. So, app is started (needed angular modules are loaded), wait for the minute and click on page that require new angular module, it failed.
In browser console I see the following:
`ERROR Error: Uncaught (in promise): Error: (SystemJS) XHR error (500 Internal Server Error) loading http://example.com/app/account/account.module.js`

IIS Express "Logs" folder I see this:
`2017-07-03 13:20:40 127.0.0.1 GET /app/account/account.module.js - 80 - 127.0.0.1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/59.0.3071.115+Safari/537.36 http://example.com/ 500 0 0 59`

In "TraceLogFiles", I also don't see anything useful about my error.

**Comment by Tratcher** [comment]
I meant application logs: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging

**Comment by gladkiy** [comment]
I think I found out what's the reason of that error. Unfortunatelly Logger didn't help me, but VS helped. 
This is error - https://www.screencast.com/t/gllPne2r

So I have app with dynamic db connection strings.

In **Startup.cs ConfigureServices** method I have the following rows:
```
services.AddDbContext<ApplicationContext>(options => options.UseSqlServer(Configuration.GetConnectionString("ApplicationContext"), builder => builder.MigrationsAssembly("AC.DBModels")));
            services.AddDbContext<DataContext>(options => options.UseSqlServer(Configuration.GetConnectionString("DataContext"), builder => builder.MigrationsAssembly("AC.DBModels"))); //connection string with empty db name

            services.AddScoped(provider => DbNameExtension.Get(provider)); // getting db name from url
```

Inside **DbNameExtension.Get** method I did checking if request doesn't have extension then get db name else empty string and in that moment HttpContaxt.User.Identity.Name is null. So I think, I get db name and then AspNetCoreIdentity go to the db to check / auth user, but db name is empty string and however the current request path is something like that: http://example.com/dashboard.module.js.
So authentication was failed and app throw 500 error. Am I right? Correct me if I wrong.

**Comment by blowdart** [comment]
It sounds like it yes, if you're not resolving the DB, its going to barf :)
